encfs/RawFileIO.cpp

```diff
@@ -219,37 +219,44 @@ ssize_t RawFileIO::write(const IORequest &req) {
   rAssert(fd >= 0);
   rAssert(canWrite);
 
-  int retrys = 10;
+  // int retrys = 10;
   void *buf = req.data;
   ssize_t bytes = req.dataLen;
   off_t offset = req.offset;
 
   int eno = 0;
-  while ((bytes != 0) && retrys > 0) {
-    errno = 0;
+  /*
+   * Let's write while pwrite() writes, to avoid writing only a part of the request,
+   * whereas it could have been fully written. This to avoid inconsistencies / corruption.
+   */
+  // while ((bytes != 0) && retrys > 0) {
+  while (bytes != 0) {
     ssize_t writeSize = ::pwrite(fd, buf, bytes, offset);
     eno = errno;
```

Comment:
The changes here did not fix errno handling.  This change is making matters worse IMO.  pwrite() == 0 is not a problem.  Errno is still being mutated and handled improperly.